app/models/piece.rb

```diff
@@ -3,56 +3,41 @@ class Piece < ActiveRecord::Base
   belongs_to :user
   belongs_to :game
 
-def obstructed_move?(x, y)
-  current_row = self.row_position
-  current_col = self.col_position
-
-  # check if piece is blocking the path in positive vertical direction
-  if current_col == y && x > current_row
-    for i in (current_row+1...x) do
-      if self.game.piece_exists?(i, y)
-        return true
+  # is_obstructed? determines if there are obstructions between two squares. An obstruction is “something between point A and point B”.
+  # Raise an Error Message if Not diagnal, horizontal, or vertical. (Sample is Knight)
+  # Return true if there is a piece found between current x/y to target x/y else return false
+  # Raise an error is target is not vertical, horizontal, and diagonal.
+  def is_obstructed?(x, y)
+    current_row = self.row_position
+    current_col = self.col_position
+
+    # check if piece is blocking the path in positive vertical direction
+    if current_col == y && x > current_row
+      for i in (current_row + 1...x)
+        return game.occupied?(i, y)
```

Comment:
Hey Aaron, can you give another hint why (1,4)needs to be treated differently in your example? Is it because this method actually moves the piece to the position so (1,4) will always be occupied and will return false even though it should be true?